examples/vae/train_vae.py

```diff
@@ -0,0 +1,546 @@
+"""
+TODO: fix training mixed precision -- issue with AdamW optimizer
+"""
+
+import argparse
+import logging
+import math
+import os
+import random
+from pathlib import Path
+
+import accelerate
+import datasets
+import numpy as np
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+import torchvision
+import transformers
+from accelerate import Accelerator
+from accelerate.logging import get_logger
+from accelerate.state import AcceleratorState
+from accelerate.utils import ProjectConfiguration, set_seed
+from datasets import load_dataset
+from huggingface_hub import create_repo, upload_folder
+from packaging import version
+from torchvision import transforms
+from tqdm.auto import tqdm
+
+import diffusers
+from diffusers import AutoencoderKL
+from diffusers.optimization import get_scheduler
+from diffusers.training_utils import EMAModel
+from diffusers.utils import check_min_version, deprecate, is_wandb_available
+from diffusers.utils.import_utils import is_xformers_available
+
+import lpips
+
+if is_wandb_available():
+    import wandb
+
+logger = get_logger(__name__, log_level="INFO")
+
+
+def log_validation(test_dataloader, vae, accelerator, weight_dtype, epoch):
+    logger.info("Running validation... ")
+
+    vae_model = accelerator.unwrap_model(vae)
+    images = []
+    for _, sample in enumerate(test_dataloader):
+        x = sample["pixel_values"].to(weight_dtype)
+        reconstructions = vae_model(x).sample
+        images.append(
+            torch.cat([sample["pixel_values"].cpu(), reconstructions.cpu()], axis=0)
+        )
+
+    for tracker in accelerator.trackers:
+        if tracker.name == "tensorboard":
+            np_images = np.stack([np.asarray(img) for img in images])
+            tracker.writer.add_images(
+                "Original (left) / Reconstruction (right)", np_images, epoch
```

Comment:
change file name to be compatible with Windows
```suggestion           
                "Original (left)-Reconstruction (right)", np_images, epoch
```